src/server/middleware/request-security.middleware.ts

```
@@ -0,0 +1,236 @@
+    const maxRequestSize = this.config.maxRequestSize;
+
+    // Override event listeners to intercept 'data' events
+    req.on = function (event: string | symbol, listener: GenericEventListener) {
+      if (event === 'data') {
+        // Wrap the data listener to track cumulative size
+        const wrappedListener: DataEventListener = (chunk: Buffer | string) => {
+          if (sizeLimitExceeded) {
+            return; // Don't process more data if limit already exceeded
+          }
+
+          const chunkSize = Buffer.isBuffer(chunk) ? chunk.length : Buffer.byteLength(chunk, 'utf8');
```

ESLint

[error] Replace `·?·chunk.length` with `⏎············?·chunk.length⏎···········`

(prettier/prettier)